plotAllMCFlavor.py

```diff
@@ -851,16 +1178,49 @@ def plotDataMC(args,plot_mu,plot_el, category):
 			if datael.GetBinContent(i) == 0: continue
 			if datamu.GetBinContent(i) == 0: continue
 			dataval = datamu.GetBinContent(i)*1.0/datael.GetBinContent(i)
-			dataerr = math.sqrt((datamu.GetBinError(i)/datamu.GetBinContent(i))**2+(datael.GetBinError(i)/datael.GetBinContent(i)))
+			dataerr = math.sqrt((datamu.GetBinError(i)/datamu.GetBinContent(i))**2+(datael.GetBinError(i)/datael.GetBinContent(i))**2+scale_mu[i-1]/(datamu.GetBinContent(i)**2)+scale_el[i-1]/(datael.GetBinContent(i)**2))
 			if hhel.GetBinContent(i) == 0: continue
 			if hhmu.GetBinContent(i) == 0: continue
 			MCval = hhmu.GetBinContent(i)*1.0/hhel.GetBinContent(i)
 			MCerr = math.sqrt((errel[i-1]**0.5/hhel.GetBinContent(i))**2+(errmu[i-1]**0.5/hhmu.GetBinContent(i))**2+(hhmu.GetBinError(i)/hhmu.GetBinContent(i))**2+(hhel.GetBinError(i)/hhel.GetBinContent(i))**2)
 			yval = dataval/MCval
 			yerr=yval*math.sqrt((MCerr)**2+(dataerr)**2)
-			ratioDataMC.SetPoint(i, xval, yval)
 			ratioDataMC.SetPointError(i, xerr, xerr, yerr, yerr)
-			print ("M = %f, r+-e = %f +- %f"%(xval, yval, abs(yval-1)/yerr))
+                        print ("M = %f, r+-e = %f +- %f"%(xval, yval, yerr))
+	if args.alter:
+		ers=0
+                dmCanvas = TCanvas("hCanvas", "Distribution", 800,800)
+                ratioDataMC = ROOT.TGraphAsymmErrors(datamu.GetSize()-2)
+		for i in range(1, datamu.GetSize()-1):
+			xval=datamu.GetBinCenter(i)
+			fval=func.Eval(xval)
+			print "At the bin center with the mass %f, scaling function value is %f, dimuon event yield is %f, dielectron event yield is %f"%(datamu.GetBinCenter(i), fval, datamu.GetBinContent(i)*datamu.GetBinWidth(i), datael.GetBinContent(i)*datael.GetBinWidth(i))
+			muval=datamu.GetBinContent(i)/func.Eval(xval)
+			ParErr0=func.GetParError(0)
+			ParErr1=func.GetParError(1)
+			ParErr2=func.GetParError(2)
+			fErr=math.sqrt(ParErr0**2+(ParErr1*math.log(xval))**2+(ParErr2*xval**2)**2)
+			#fErr=0
+			#err=math.sqrt(datamu.GetBinError(i)**2+scale_mu[i-1]+muval**2*fErr**2)/fval
+			err=muval*fErr/fval
+			datamu.SetBinContent(i,muval)
+			datamu.SetBinError(i,err)

```

Comment:
Is this from one our tests? It looks like you are are using the fit errors only here.